src/main/java/net/masterthought/cucumber/reducers/ReducingMethod.java

```diff
@@ -13,6 +13,34 @@
      */
     MERGE_FEATURES_BY_ID,
 
+    /**
+     * Merge features and scenarios from different JSON files of different runs
+     * into a single report by features' and scenarios' ids.
+     *
+     * Merging rules:
+     * - Every new feature which is not in the result list is appended to the end.
+     *
+     * - When the results list already has a feature with such Id then we go down and apply the rules below to the scenarios:
+     *
+     *      1. if there is no scenario with a given Id in the feature's elements list
+     *         then add the scenario to the end of the list.
+     *
+     *      2. if scenario has a background (which is a previous element in the elements list)
+     *        then both elements are added to the end of the current feature's elements list.
+     *
+     *      3. if there is a scenario with a given Id then:
+     *          scenario + background case: replace both elements (existing element with Id and its background with new ones)
+     *          scenario only: replace only given scenario by index in the array.
+     *
+     * Example:
+     * Original cucumber report is "cucumber.json". Let's look a situation when couple of tests failed there.
+     * Cucumber runner generates a new report, for example, cucumber-rerun.json as a result of rerun the failed tests.
+     *
+     * In that case you will have a merged report where all failed tests from the original cucumber.json file
+     * are overridden with the results from the cucumber-rerun.json.
+     */
+    MERGE_FEATURES_WITH_RETEST,
```

Comment:
will you provide support for that later in https://github.com/jenkinsci/cucumber-reports-plugin ?

Reply:
:) will you merge the PR if I say "No"?
don't worry) I'll add this feature to the plugin. I just didn't know about it.

Reply:
Once this PR is merged I will publish 5.0 after some manual tests